### ISSUE TITLE
controller and builder services fail to start after node reboot

### ISSUE THREAD

**Issue by bilts** [deis/deis#2293]
I've seen this issue multiple times, both against 0.13 and now against 0.14.1 (I haven't tried 0.15, but I don't see anything there which would fix it).  I've seen it in both AWS and Vagrant.  

When I shut down the controller service's node (the registry service has been on the same node every time, fwiw), the controller service fails to come back up, as does the builder service which I believe is stuck in a loop waiting for the controller service.  This looks similar to a couple of other issues (#2235 #2132 #2146) except that my version is up-to-date with the recommendations, and my `/deis/platform/domain` key is set.

The controller service is in an inactive/dead state and does not seem to attempt to start when I run `deisctl start controller`.  I get no output, no journal, and the command hangs indefinitely.

This has been going on for around 20 hours, so I don't expect it to ever recover.

Some relevant logs and debug info:

```
$ deisctl list
UNIT                MACHINE         LOAD    ACTIVE      SUB
deis-builder.service        b5083ca9.../10.21.1.44  loaded  activating  start-post
deis-cache.service      b5083ca9.../10.21.1.44  loaded  active      running
deis-controller.service     b5083ca9.../10.21.1.44  loaded  inactive    dead
deis-database.service       b5083ca9.../10.21.1.44  loaded  active      running
deis-logger-data.service    b5083ca9.../10.21.1.44  loaded  active      exited
deis-logger.service     b5083ca9.../10.21.1.44  loaded  active      running
deis-logspout.service       2fa78f14.../10.21.1.45  loaded  active      running
deis-logspout.service       b5083ca9.../10.21.1.44  loaded  active      running
deis-logspout.service       b5420254.../10.21.2.243 loaded  active      running
deis-publisher.service      2fa78f14.../10.21.1.45  loaded  active      running
deis-publisher.service      b5083ca9.../10.21.1.44  loaded  active      running
deis-publisher.service      b5420254.../10.21.2.243 loaded  active      running
deis-registry.service       b5083ca9.../10.21.1.44  loaded  active      running
deis-router@1.service       b5420254.../10.21.2.243 loaded  active      running
deis-router@2.service       2fa78f14.../10.21.1.45  loaded  active      running
deis-router@3.service       b5083ca9.../10.21.1.44  loaded  active      running
deis-store-daemon.service   2fa78f14.../10.21.1.45  loaded  active      running
deis-store-daemon.service   b5083ca9.../10.21.1.44  loaded  active      running
deis-store-daemon.service   b5420254.../10.21.2.243 loaded  active      running
deis-store-gateway.service  b5083ca9.../10.21.1.44  loaded  active      running
deis-store-monitor.service  2fa78f14.../10.21.1.45  loaded  active      running
deis-store-monitor.service  b5083ca9.../10.21.1.44  loaded  active      running
deis-store-monitor.service  b5420254.../10.21.2.243 loaded  active      running
```

I fiddled with conf.toml to allow verbose output on the following:

```
$ deisctl journal builder
[repeats many times]
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /home/git/.ssh/authorized_keys in sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /home/git/builder out of sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: ERROR Config check failed: exit status 1
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /home/git/check-repos in sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /usr/local/bin/gitreceive in sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /usr/local/bin/push-images in sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: INFO Target config /home/git/receiver out of sync
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:33Z c031bba113db confd[2896]: ERROR Config check failed: exit status 1
Oct 28 16:04:33 ip-10-21-1-44.ec2.internal sh[6132]: builder: waiting for confd to write initial templates...
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /home/git/.ssh/authorized_keys in sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /home/git/builder out of sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: ERROR Config check failed: exit status 1
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /home/git/check-repos in sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /usr/local/bin/gitreceive in sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /usr/local/bin/push-images in sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: INFO Target config /home/git/receiver out of sync
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: 2014-10-28T16:04:38Z c031bba113db confd[2911]: ERROR Config check failed: exit status 1
Oct 28 16:04:38 ip-10-21-1-44.ec2.internal sh[6132]: builder: waiting for confd to write initial templates...
[repeats many times]
```

All builder dependencies, including controller seem to have keys:

```
$ etcdctl ls --recursive
/deis
/deis/platform
/deis/platform/domain
/deis/router
/deis/router/10.21.1.44
/deis/router/10.21.1.44/host
/deis/router/10.21.1.44/port
/deis/router/10.21.2.243
/deis/router/10.21.2.243/port
/deis/router/10.21.2.243/host
/deis/router/gzipHttpVersion
/deis/router/gzipMinLength
/deis/router/gzipTypes
/deis/router/10.21.1.45
/deis/router/10.21.1.45/port
/deis/router/10.21.1.45/host
/deis/router/gzip
/deis/router/gzipCompLevel
/deis/router/gzipDisable
/deis/router/gzipProxied
/deis/router/gzipVary
/deis/store
/deis/store/gatewayKeyring
/deis/store/hosts
/deis/store/hosts/10.21.1.44
/deis/store/hosts/10.21.1.45
/deis/store/hosts/10.21.2.243
/deis/store/pgNum
/deis/store/delayStart
/deis/store/fsid
/deis/store/gateway
/deis/store/gateway/accessKey
/deis/store/gateway/port
/deis/store/gateway/host
/deis/store/gateway/secretKey
/deis/store/minSize
/deis/store/monKeyring
/deis/store/monSetupComplete
/deis/store/monSetupLock
/deis/store/osds
/deis/store/osds/10.21.2.243
/deis/store/osds/10.21.1.44
/deis/store/osds/10.21.1.45
/deis/store/adminKeyring
/deis/store/size
/deis/builder
/deis/builder/users
/deis/builder/users/bilts
/deis/builder/users/bilts/73:5b:[redacted]
/deis/cache
/deis/cache/host
/deis/cache/port
/deis/controller
/deis/controller/registrationEnabled
/deis/controller/secretKey
/deis/controller/webEnabled
/deis/controller/builderKey
/deis/controller/protocol
/deis/database
/deis/database/adminUser
/deis/database/engine
/deis/database/host
/deis/database/name
/deis/database/adminPass
/deis/database/bucketName
/deis/database/password
/deis/database/port
/deis/database/user
/deis/domains
/deis/logs
/deis/logs/host
/deis/logs/port
/deis/registry
/deis/registry/secretKey
/deis/registry/bucketName
/deis/registry/host
/deis/registry/port
/deis/registry/protocol
/deis/services
/deis/services/example-ruby-sinatra
/deis/services/example-ruby-sinatra/example-ruby-sinatra_v6.web.1
```

controller journal.  Note I have tried to start it a couple of times and a couple of different ways since the last journaled message, and let it run for hours.  I never get any output or journal messages.

```
$ deisctl journal controller
-- Logs begin at Mon 2014-10-27 16:22:37 UTC. --
Oct 27 19:28:36 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "DELETE /v1/apps/example-ruby-sinatra HTTP/1.0" 204 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:28:41 ip-10-21-1-44.ec2.internal sh[2029]: INFO marble-original: config marble-original-8295c4a updated
Oct 27 19:28:41 ip-10-21-1-44.ec2.internal sh[2029]: INFO marble-original: release marble-original-v1 created
Oct 27 19:28:41 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "POST /v1/apps HTTP/1.0" 201 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:28:49 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "DELETE /v1/apps/marble-original HTTP/1.0" 204 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:28:56 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: config example-ruby-sinatra-725e194 updated
Oct 27 19:28:56 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v1 created
Oct 27 19:28:56 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "POST /v1/apps HTTP/1.0" 201 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:29:06 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/push HTTP/1.1" 201 - "curl/7.35.0"
Oct 27 19:29:06 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/config HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:30:26 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: build example-ruby-sinatra-045ead1 created
Oct 27 19:30:26 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v2 created
Oct 27 19:30:28 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: pquinn scaled containers web=1
Oct 27 19:31:02 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/build HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:34:41 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/push HTTP/1.1" 201 - "curl/7.35.0"
Oct 27 19:34:41 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/config HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:35:59 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: build example-ruby-sinatra-b045927 created
Oct 27 19:35:59 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v3 created
Oct 27 19:36:43 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/build HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:36:56 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "GET /v1/apps/example-ruby-sinatra/logs HTTP/1.0" 200 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:37:43 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/push HTTP/1.1" 201 - "curl/7.35.0"
Oct 27 19:37:43 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/config HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:38:59 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: build example-ruby-sinatra-5788149 created
Oct 27 19:38:59 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v4 created
Oct 27 19:39:40 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/build HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:42:02 ip-10-21-1-44.ec2.internal sh[2029]: 172.17.42.1 "GET /v1/apps/example-ruby-sinatra/logs HTTP/1.0" 200 - "python-requests/2.3.0 CPython/2.7.5 Darwin/12.5.0"
Oct 27 19:43:11 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/push HTTP/1.1" 201 - "curl/7.35.0"
Oct 27 19:43:11 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/config HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:44:28 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: build example-ruby-sinatra-f74f646 created
Oct 27 19:44:28 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v5 created
Oct 27 19:45:10 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/build HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:46:17 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/push HTTP/1.1" 201 - "curl/7.35.0"
Oct 27 19:46:17 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/config HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 19:47:35 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: build example-ruby-sinatra-ce6f1fa created
Oct 27 19:47:35 ip-10-21-1-44.ec2.internal sh[2029]: INFO example-ruby-sinatra: release example-ruby-sinatra-v6 created
Oct 27 19:48:08 ip-10-21-1-44.ec2.internal sh[2029]: 10.21.1.45 "POST /v1/hooks/build HTTP/1.1" 200 - "curl/7.35.0"
Oct 27 20:15:31 ip-10-21-1-44.ec2.internal sh[2029]: 2014/10/27 20:15:31 Post http:///var/run/docker.sock/v1.15/containers/6140f2546f83fc41a66fc925a6832b7ea8b0474d8a8d0910bb4683c2b45ef1ec/wait: read unix /var/run/docker.sock: connection reset by peer
Oct 27 20:15:32 ip-10-21-1-44.ec2.internal docker[31780]: write unix /var/run/docker.sock: broken pipe
Oct 27 20:15:32 ip-10-21-1-44.ec2.internal docker[31780]: 2014/10/27 20:15:32 Error: failed to remove one or more containers
Oct 27 20:15:32 ip-10-21-1-44.ec2.internal systemd[1]: Stopped deis-controller.
```

Please let me know if any other information would help in debugging this

**Comment by fagiani** [comment]
Hey @carmstrong,

I was able to reproduce this very issue in the version 1.4.1

Unfortunately, even after trying all the restarts suggested here the status of the controller and logger stay as dead and logspout stays as auto-restart status.

As there are no logs, it is impossible to understand what is going on.

Dou you have any idea on how to fix it?

Thanks!

**Comment by carmstrong** [comment]
@fagiani This is during a platform start, right? That makes sense that controller and logger are dead, as logspout starts early in the start cycle. Can you open a new issue with the output of `journalctl -fu deis-logspout` on the host that's having issues?

**Comment by fagiani** [comment]
@carmstrong thanks for the prompt response!

The issue happened with deis working fine and not on platform start. It was right after a git push of one of the apps.

As we are moving on to production environment, after several trial and error, we restarted the 3 deis hosts and somehow cloud formation has terminated them and recreated other hosts and after that, the whole thing stopped working.

We have then started from scratch therefore I am not able to open another issue unless it happens again. At this point, I guess I just want to make sure you guys are aware this very issue still happens somehow in the latest version.

I'll let you know if we face that again!
